src/SixLabors.Fonts/Tables/AdvancedTypographic/CoverageTable.cs

```diff
@@ -0,0 +1,164 @@
+            return coverageFormat switch
+            {
+                1 => CoverageFormat1Table.Load(reader),
+                2 => CoverageFormat2Table.Load(reader),
+                _ => throw new InvalidFontFileException($"Invalid value for 'coverageFormat' {coverageFormat}. Should be '1' or '2'.")
+            };
+        }
+
+        public static CoverageTable[] LoadArray(BigEndianBinaryReader reader, long offset, ushort[] coverageOffsets)
+        {
+            var tables = new CoverageTable[coverageOffsets.Length];
+            for (int i = 0; i < tables.Length; i++)
+            {
+                tables[i] = Load(reader, offset + coverageOffsets[i]);
+            }
+
+            return tables;
+        }
+    }
+
+    internal sealed class CoverageFormat1Table : CoverageTable
+    {
+        private readonly ushort[] glyphArray;
+
+        private CoverageFormat1Table(ushort[] glyphArray)
+            => this.glyphArray = glyphArray;
+
+        public override int CoverageIndexOf(ushort glyphId)
+        {
+            int n = Array.BinarySearch(this.glyphArray, glyphId);
+            return n < 0 ? -1 : n;
+        }
+
+        public override IEnumerator<ushort> GetEnumerator()
+            => ((IEnumerable<ushort>)this.glyphArray).GetEnumerator();
+
+        public static CoverageFormat1Table Load(BigEndianBinaryReader reader)
+        {
+            // +--------+------------------------+-----------------------------------------+
+            // | Type   | Name                   | Description                             |
+            // +========+========================+=========================================+
+            // | uint16 | coverageFormat         | Format identifier — format = 1          |
+            // +--------+------------------------+-----------------------------------------+
+            // | uint16 | glyphCount             | Number of glyphs in the glyph array     |
+            // +--------+------------------------+-----------------------------------------+
+            // | uint16 | glyphArray[glyphCount] | Array of glyph IDs — in numerical order |
+            // +--------+------------------------+-----------------------------------------+
+            ushort glyphCount = reader.ReadUInt16();
+            ushort[] glyphArray = reader.ReadUInt16Array(glyphCount);
+
+            return new CoverageFormat1Table(glyphArray);
+        }
+    }
+
+    internal sealed class CoverageFormat2Table : CoverageTable
+    {
+        private readonly CoverageRangeRecord[] records;
+
+        private CoverageFormat2Table(CoverageRangeRecord[] records)
+            => this.records = records;
+
+        public override int CoverageIndexOf(ushort glyphId)
+        {
+            for (int i = 0; i < this.records.Length; i++)
+            {
+                CoverageRangeRecord rec = this.records[i];
+                if (rec.StartGlyphId <= glyphId && glyphId <= rec.EndGlyphId)
```

Comment:
Is this maybe a mistake here and the comparison should be flipped?

```suggestion
                if (rec.StartGlyphId >= glyphId && glyphId <= rec.EndGlyphId)
```

Reply:
Ah no, the glyph has to be within the start and end of the range (inclusive).